src/__benchmarks__/schema-nested-arrays.benchmark.ts

```diff
@@ -0,0 +1,277 @@
+
+  const BlogAuthor = new GraphQLObjectType({
+    name: "Author",
+    fields: () => ({
+      id: {
+        type: GraphQLString,
+        resolve: author => Promise.resolve(author.id)
+      },
+      name: {
+        type: GraphQLString,
+        resolve: author => Promise.resolve(author.name)
+      },
+      pic: {
+        args: { width: { type: GraphQLInt }, height: { type: GraphQLInt } },
+        type: BlogImage,
+        resolve: (obj, { width, height }) => obj.pic(width, height)
+      },
+      articles: {
+        type: new GraphQLList(BlogArticle),
+        resolve: _ => Promise.resolve(articles)
+      }
+    })
+  });
+
+  const BlogArticleBadge: GraphQLObjectType = new GraphQLObjectType({
+    name: "ArticleBadge",
+    fields: {
+      color: {
+        type: GraphQLString,
+        resolve: badge => Promise.resolve(badge && badge.color)
+      },
+      text: {
+        type: GraphQLString,
+        resolve: badge => Promise.resolve(badge && badge.text)
+      }
+    }
+  });
+
+  const BlogArticleAdvert: GraphQLObjectType = new GraphQLObjectType({
+    name: "ArticleAdvert",
+    fields: {
+      text: {
+        type: GraphQLString,
+        resolve: advert => Promise.resolve(advert && advert.text)
+      },
+      image: {
+        type: BlogImage,
+        resolve: advert => Promise.resolve(advert && advert.image)
+      }
+    }
+  });
+
+  const BlogArticle: GraphQLObjectType = new GraphQLObjectType({
+    name: "Article",
+    fields: {
+      id: {
+        type: new GraphQLNonNull(GraphQLID),
+        resolve: article => Promise.resolve(article.id)
+      },
+      isPublished: {
+        type: GraphQLBoolean,
+        resolve: article => Promise.resolve(article.isPublished)
+      },
+      author: { type: BlogAuthor },
+      title: {
+        type: GraphQLString,
+        resolve: article => Promise.resolve(article && article.title)
+      },
+      body: {
+        type: GraphQLString,
+        resolve: article => Promise.resolve(article.body)
+      },
+      keywords: {
+        type: new GraphQLList(GraphQLString),
+        resolve: article => Promise.resolve(article.keywords)
+      },
+      badges: {
+        type: new GraphQLList(BlogArticleBadge)
+      },
+      adverts: {
+        type: new GraphQLList(BlogArticleAdvert)
+      }
+    }
+  });
+
+  const BlogQuery = new GraphQLObjectType({
+    name: "Query",
+    fields: {
+      article: {
+        type: BlogArticle,
+        args: { id: { type: GraphQLID } },
+        resolve: (_, { id }) => article(id)
+      },
+      feed: {
+        type: new GraphQLList(BlogArticle),
+        resolve: () =>
+          Promise.resolve([
+            article(1),
+            article(2),
+            article(3),
+            article(4),
+            article(5),
+            article(6),
+            article(7),
+            article(8),
+            article(9),
+            article(10)
+          ])
+      }
+    }
+  });
+
+  for (let i = 0; i < 25; i++) {
```

Comment:
Can we extract this 25 to a `const` at the top of the file to be easy to change the size?

Reply:
sure, sorry about that, done